### PR TITLE
Bluetooth: btp: Add initial commands for Mesh service

### PR DESCRIPTION
**Pull request by sjanc**

### CODE REVIEW

File: tests/bluetooth/tester/btp_spec.txt [hunk 1]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+					Output Size (1 octet)
+					Output Actions (2 octets)
+					Input Size (1 octet)
+					Input Actions (2 octets)
+
+		This command is used to provide provisioning configuration for
+		node.
+
+		Output actions is a bitmask with following available bits:
+			1	Blink
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: Would be better to start with bit 0

File: tests/bluetooth/tester/btp_spec.txt [hunk 2]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+			2	Beep
+			3	Vibrate
+			4	Display number
+			5	Display string
+
+		Input actions is a bitmask with following available bits:
+			1	Push
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: ditto, start with 0 bit

File: tests/bluetooth/tester/btp_spec.txt [hunk 3]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x02 - Init command/response
+		Controller Index:	<controller id>
+		Command parameters:	Static Auth (16 octets)
+					Output Size (1 octet)
+					Output Actions (2 octets)
+					Input Size (1 octet)
+					Input Actions (2 octets)
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: Mention about response parameters, even if <none> to be consistent

File: tests/bluetooth/tester/btp_spec.txt [hunk 4]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+			1	Push
+			2	Twist
+			3	Enter number
+			5	Enter string
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x03 - Reset command/response
+		Controller Index:	<controller id>
+		Command parameters:	<none>
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: Mention about response parameters, even if <none> to be consistent

File: tests/bluetooth/tester/btp_spec.txt [hunk 5]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+
+		This command is used to reset state of a provisioned node making
+		it to start sending unprovisioned beacons.
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x04 - Input number command/response
+		Controller Index:	<controller id>
+		Command parameters:	Number (4 octets)
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: Mention about response parameters, even if <none> to be consistent

File: tests/bluetooth/tester/btp_spec.txt [hunk 6]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+		This command is used to reply with input number as a response to
+		Input action event.
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x05 - Input string command/response
+		Controller Index:	<controller id>
+		Command parameters:	String Len (1 octet)
+					String (variable)
+
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: Mention about response parameters, even if <none> to be consistent

File: tests/bluetooth/tester/btp_spec.txt [hunk 7]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+		be set to 0. If specific bit is not present in response (less
+		than required bytes received) it shall be assumed that command
+		is not supported.
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x02 - Init command/response
+		Controller Index:	<controller id>
+		Command parameters:	Static Auth (16 octets)
+					Output Size (1 octet)
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: Not sure if I get it. Why cannot use bitmask to specify multiple actions? Is output size parameter necesary? The same question applies to Input size/actions

File: tests/bluetooth/tester/btp_spec.txt [hunk 8]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+					Output Actions (2 octets)
+					Input Size (1 octet)
+					Input Actions (2 octets)
+
+		This command is used to provide provisioning configuration for
+		node.
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: Please provide info about Static Auth parameter.

File: tests/bluetooth/tester/btp_spec.txt [hunk 9]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+
+		Depending on action either number or string fields are valid.
+
+	Opcode 0x81 - Input action Event
+
+		Controller Index:	<controller id>
+		Event parameters:	Action (2 octets)
+					Size (1 octet)
+
+		This event indicates that input actions is required. Depending
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: typo "actions"->"action"

File: tests/bluetooth/tester/btp_spec.txt [hunk 10]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+	Opcode 0x03 - Reset command/response
+		Controller Index:	<controller id>
+		Command parameters:	<none>
+
+		This command is used to reset state of a provisioned node making
+		it to start sending unprovisioned beacons.
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x04 - Input number command/response
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: You could merge "Input number command/response" and "Input string command/response" into one command and add action field, just like you did in "Input action Event"

File: tests/bluetooth/tester/btp_spec.txt [hunk 11]
```diff
@@ -1277,3 +1277,109 @@ Events:
...
+			2	Beep
+			3	Vibrate
+			4	Display number
+			5	Display string
+
+		Input actions is a bitmask with following available bits:
+			1	Push
+			2	Twist
+			3	Enter number
+			5	Enter string
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: what happened with 4th bit?

**Review by MariuszSkamra — CHANGES_REQUESTED** [review]

**Comment by MariuszSkamra** [comment]
Don not forget to add dev UUID parameter to init command

**Comment by MariuszSkamra** [comment]
Looks good for me. If there will be some additional commands needed, we can always extend BTP.

**Comment by MariuszSkamra** [comment]
@jhedberg I think document is ready to be merged if OK for you

File: tests/bluetooth/tester/btp_spec.txt [hunk 12]
```diff
@@ -1289,3 +1289,136 @@ Events:
...
+
+	Opcode 0x03 - Provision Node command/response
+		Controller Index:	<controller id>
+		Command parameters:	Network Key (16 octets)
+					Network Key Index (2 octets)
+					Provisioning Flags (1 octet)
+					IV Index (4 octets)
+					Sequence Number (4 octets)
+					Primary Element Address (2 octets)
+					Device Key (2 octets)
```
Comment (MariuszSkamra) [inline tests/bluetooth/tester/btp_spec.txt]: Device Key is 16 octets

**Review by MariuszSkamra — CHANGES_REQUESTED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by MariuszSkamra — APPROVED** [review]

**Pull request merged**